runtime/v2/bundle.go

```diff
@@ -59,22 +59,35 @@ func NewBundle(ctx context.Context, root, state, id string, spec []byte) (b *Bun
 		Namespace: ns,
 	}
 	paths := []string{b.Path, work}
-	// create base directories
-	for _, d := range paths {
-		if err := os.MkdirAll(filepath.Dir(d), 0711); err != nil {
-			return nil, err
-		}
-		if err := os.Mkdir(d, 0711); err != nil {
-			return nil, err
-		}
-	}
 	defer func() {
 		if err != nil {
 			for _, d := range paths {
 				os.RemoveAll(d)
 			}
 		}
 	}()
+	// create state directory for the bundle
+	if err := os.MkdirAll(filepath.Dir(b.Path), 0711); err != nil {
+		return nil, err
+	}
+	if err := os.Mkdir(b.Path, 0711); err != nil {
+		return nil, err
+	}
+	paths = append(paths, b.Path)
```

Comment:
paths was already initialized with this, is `paths` supposed to start off empty now?

Reply:
empty